src/nft/components/bag/BagFooter.tsx

```diff
@@ -155,11 +279,51 @@ export const BagFooter = ({ totalEthPrice, bagStatus, fetchAssets, eventProperti
     return parseEther(balanceInEth).gte(totalEthPrice)
   }, [connected, chainId, balanceInEth, totalEthPrice])
 
-  const { buttonText, disabled, warningText, handleClick } = useMemo(() => {
+  const isPending = PENDING_BAG_STATUSES.includes(bagStatus)
+  const activeCurrency = inputCurrency ?? defaultCurrency
+  const usingPayWithAnyToken = !!inputCurrency && shouldUsePayWithAnyToken
+
+  const parsedOutputAmount = useMemo(() => {
+    return tryParseCurrencyAmount(formatEther(totalEthPrice.toString()), defaultCurrency ?? undefined)
+  }, [defaultCurrency, totalEthPrice])
+  const { state: tradeState, trade, maximumAmountIn } = usePayWithAnyTokenSwap(inputCurrency, parsedOutputAmount)
+  const { allowance, isAllowancePending, isApprovalLoading, updateAllowance } = usePermit2Approval(
+    trade?.inputAmount.currency.isToken ? (trade?.inputAmount as CurrencyAmount<Token>) : undefined,
+    maximumAmountIn
+  )
+
+  const fiatValueTradeInput = useStablecoinValue(trade?.inputAmount)
+  const fiatValueTradeOutput = useStablecoinValue(parsedOutputAmount)
+  const usdcValue = usingPayWithAnyToken ? fiatValueTradeInput : fiatValueTradeOutput
+  const stablecoinPriceImpact = useMemo(
+    () =>
+      tradeState === TradeState.SYNCING || !trade
+        ? undefined
+        : computeFiatValuePriceImpact(fiatValueTradeInput, fiatValueTradeOutput),
+    [fiatValueTradeInput, fiatValueTradeOutput, tradeState, trade]
+  )
+  const { priceImpactWarning, priceImpactColor } = useMemo(() => {
+    const severity = warningSeverity(stablecoinPriceImpact)
+
+    if (severity < 1) {
+      return { priceImpactWarning: false, priceImpactColor: undefined }
+    }
+
+    if (severity < 3) {
```

Comment:
nit: could set consts like LOW_IMPACT_THRESHOLD, MEDIUM_IMPACT_THRESHOLD to make values more clear